shell/browser/mac/in_app_purchase_observer.mm

```diff
@@ -94,6 +94,37 @@ - (NSString*)dateToISOString:(NSDate*)date {
   return [dateFormatter stringFromDate:date];
 }
 
+/**
+ * Convert a SKPaymentDiscount object to a PaymentDiscount structure.
+ *
+ * @param paymentDiscount - The SKPaymentDiscount object to convert.
+ */
+- (in_app_purchase::PaymentDiscount)skPaymentDiscountToStruct:
+    (SKPaymentDiscount*)paymentDiscount API_AVAILABLE(macosx(10.14.4)) {
+  in_app_purchase::PaymentDiscount paymentDiscountStruct;
+
+  if (paymentDiscount.identifier != nil) {
+    paymentDiscountStruct.identifier = [paymentDiscount.identifier UTF8String];
+  }
+
+  if (paymentDiscount.keyIdentifier != nil) {
+    paymentDiscountStruct.keyIdentifier =
+        [paymentDiscount.keyIdentifier UTF8String];
+  }
+
+  // if (paymentDiscount.nonce != nil) {
+  //   paymentDiscountStruct.nonce = [paymentDiscount.nonce uuidString];
+  // }
```

Comment:
I am not quite sure how to get the UUID value here. 
Appreciate if I get some help here 🙏

Reply:
From [Apple docs](https://developer.apple.com/documentation/storekit/skpaymentdiscount/3043531-nonce), it is typed as UUID:

![image](https://user-images.githubusercontent.com/44967897/149259292-d2cb1162-c7d7-4064-a471-0ee9e633d691.png)

I'd imagine that we would expose it in JS in `string` type.

I wanted to figure out how I could pull the Objective C UUID and expose it as `string` in JS.
I got a syntax error trying to read the UUID value with `[paymentDiscount.nonce uuidString]`